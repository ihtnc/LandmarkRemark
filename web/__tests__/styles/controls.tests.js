import React from 'react';
import { shallow } from 'enzyme';
import {
  Wrapper,
  Header,
  Status,
  Label,
  Input,
  PasswordStrengthIcon,
  ButtonWrapper,
  Button
} from "@styles/controls";

describe("Controls styled components", () => {
  test("Wrapper should match latest snapshot", () => {
    const wrapper = shallow(<Wrapper />)
    expect(wrapper).toMatchSnapshot();
  });

  test("Header should match latest snapshot", () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot();
  });

  describe("Status", () => {
    test("should render correctly for error", () => {
      const wrapper = shallow(<Status error={true} />);
      expect(wrapper).toMatchSnapshot();
    });

    test("should render correctly for non error", () => {
      const wrapper = shallow(<Status />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  test("Label should match latest snapshot", () => {
    const wrapper = shallow(<Label />)
    expect(wrapper).toMatchSnapshot();
  });

  test("Input should match latest snapshot", () => {
    const wrapper = shallow(<Input />)
    expect(wrapper).toMatchSnapshot();
  });

  test("PasswordStrengthIcon should match latest snapshot", () => {
    const wrapper = shallow(<PasswordStrengthIcon />)
    expect(wrapper).toMatchSnapshot();
  });

  test("ButtonWrapper should match latest snapshot", () => {
    const wrapper = shallow(<ButtonWrapper />)
    expect(wrapper).toMatchSnapshot();
  });

  test("Button should match latest snapshot", () => {
    const wrapper = shallow(<Button />)
    expect(wrapper).toMatchSnapshot();
  });
});