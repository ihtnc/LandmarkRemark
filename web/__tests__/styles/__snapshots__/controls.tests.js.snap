// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls styled components Arrow should match latest snapshot 1`] = `
<StyledComponent
  direction="left"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-jzJRlG",
        "isStatic": false,
        "rules": Array [
          "
  background-image: url(",
          [Function],
          ");
  align-self: center;
  width: ",
          [Function],
          ";
  height: ",
          [Function],
          ";
  background-repeat: no-repeat;
  background-size: contain;
  display: inline-block;
  margin-top: 3px;
  margin-bottom: -3px;
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-jzJRlG",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components Button should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gZMcBi",
        "isStatic": true,
        "rules": Array [
          "
  min-width: 80px;
  padding: 1vh 1vw;
  font-size: 16px;
  align-self: center;
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-gZMcBi",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components ButtonWrapper should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-iwsKbI",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  justify-content: space-between;
  flex-direction: row-reverse;
  margin: 1vh 1vw;

  > span:last-child {
    padding: 1vh 1vw 1vh 0px;
    margin: 1vh 1vw 1vh 0px;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-iwsKbI",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components Caret should match latest snapshot 1`] = `
<StyledComponent
  expand={true}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-fjdhpX",
        "isStatic": false,
        "rules": Array [
          "
  background-image: url(",
          [Function],
          ");
  width: 20px;
  height: 20px;
  background-repeat: no-repeat;
  background-size: contain;
  display: inline-block;
  margin-right: 8px;
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-fjdhpX",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components GoogleMarker should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-jTzLTM",
        "isStatic": false,
        "rules": Array [
          "
  background-image: url(",
          [Function],
          ");
  align-self: center;
  width: ",
          [Function],
          ";
  height: ",
          [Function],
          ";
  background-repeat: no-repeat;
  background-size: contain;
  display: inline-block;
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-jTzLTM",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  size="18px"
/>
`;

exports[`Controls styled components Header should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  font-weight: bold;
  padding: 1vh 1vw;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components InlineTextWrapper should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-VigVT",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-direction: row;
  align-items: center;

  > * {
    margin: 0px;
    padding: 0px 1vw 0px 0px;
  }

  > :first-child {
    padding-left: 1vw;
  }


  *:hover {
    cursor: pointer;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-VigVT",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components Input should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": true,
        "rules": Array [
          "
  font-size: larger;
  padding: 1vh 1vw;
  margin: 1vh 1vw;
  border: 1px solid;
",
        ],
      },
      "displayName": "styled.input",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": "input",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components Label should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": true,
        "rules": Array [
          "
  padding: 1vh 1vw;
  align-self: center;
  min-width: 80px;
  width: 10vw;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components PasswordStrengthIcon should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-dnqmqq",
        "isStatic": false,
        "rules": Array [
          "
  background-image: url(",
          [Function],
          ");
  width: 30px;
  height: 30px;
  align-self: center;
  background-repeat: no-repeat;
  background-size: contain;
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-dnqmqq",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components Status should render correctly for error 1`] = `
<StyledComponent
  error={true}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  padding: 1vh 1vw;
  margin: 1vh 1vw 1vh 0px;
  color: ",
          [Function],
          ";
  align-self: flex-start;
  display: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components Status should render correctly for non error 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  padding: 1vh 1vw;
  margin: 1vh 1vw 1vh 0px;
  color: ",
          [Function],
          ";
  align-self: flex-start;
  display: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components TextWrapper should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gqjmRU",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-bottom: 1vh;

  *:hover {
    cursor: pointer;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-gqjmRU",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Controls styled components Wrapper should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  font-size: 16px;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  min-width: 200px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
